include/ygm/container/oversized_bag.hpp

```diff
@@ -0,0 +1,335 @@
+  oversized_bag(ygm::comm &comm) : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+  }
+
+  oversized_bag(ygm::comm &comm, std::initializer_list<Item> l)
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+    if (m_comm.rank0()) {
+      for (const Item &i : l) {
+        async_insert(i);
+      }
+    }
+    m_comm.barrier();
+  }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /**
+   * @brief Construct a new oversized bag object from a directory of files, arguments ect could be different
+   */
+  oversized_bag(ygm::comm &comm, std::string dir)
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+    /**
+     * @todo Implement this
+     */
+  }
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  template <typename STLContainer>
+  oversized_bag(ygm::comm          &comm,
+      const STLContainer &cont) requires detail::STLContainer<STLContainer> &&
+      std::convertible_to<typename STLContainer::value_type, Item>
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+
+    for (const Item &i : cont) {
+      this->async_insert(i);
+    }
+    m_comm.barrier();
+  }
+
+  template <typename YGMContainer>
+  oversized_bag(ygm::comm          &comm,
+      const YGMContainer &yc) requires detail::HasForAll<YGMContainer> &&
+      detail::SingleItemTuple<typename YGMContainer::for_all_args>
+      : m_comm(comm), pthis(this), partitioner(comm) {
+    pthis.check(m_comm);
+
+    yc.for_all([this](const Item &value) { this->async_insert(value); });
+
+    m_comm.barrier();
+  }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /**
+   * @todo we'll need to close all of the files we opened
+   */
+  ~oversized_bag() { m_comm.barrier(); }
+
+  /** @todo ----------------------------------------------unchanged----------------------------------------------*/
+  oversized_bag(const self_type &other)  // If I remove const it compiles
+      : m_comm(other.comm()), pthis(this), partitioner(other.comm()) {
+    pthis.check(m_comm);
+  }
+
+  oversized_bag(self_type &&other) noexcept
+      : m_comm(other.comm()),
+        pthis(this),
+        partitioner(other.comm()),
+        m_local_bag(std::move(other.m_local_bag)) {
+    pthis.check(m_comm);
+  }
+
+  oversized_bag &operator=(const self_type &other) { return *this = oversized_bag(other); }
+  /** @todo --------------------------------------------end unchanged--------------------------------------------*/
+
+  /** @todo This might be hard to reimplement, we'll need to look at if we can swap on fstream object or if we'll have to swap data
+   * then have other.open() and this->open() to reopen the insert file
+   */
+  oversized_bag &operator=(self_type &&other) noexcept {
```

Comment:
Just to make sure, this function would be responsible for swapping the content of two bags? This would mean swapping information between the files themselves, correct?

Reply:
Yes, thats correct. I have implemented it sort of in my develop branch.